packages/principal/src/index.ts

```diff
@@ -107,6 +107,14 @@ export class Principal {
     return this.toText();
   }
 
+  /**
+   * Serializes to JSON
+   * @returns {string} string
+   */
+  public toJSON(): string {
+    return this.toText();
```

Comment:
Isn't this a bit weak? It means that any string is either a principal or a string which can be quite error prone if you stringify and parse an object that contains string and principals.

In `@dfinity/utils` we prefix principal to make their encoding and decoding more safe.

https://github.com/dfinity/ic-js/blob/main/packages/utils/src/utils/json.utils.ts

Reply:
We could serialize into `{ "__principal__": principal.toText() }` instead. The main case I wanted to cover was just ensuring that `JSON.stringify()` on objects containing principals handles the case gracefully